tools/local-environment/tests/addChainCase.ts

```diff
@@ -211,6 +203,103 @@ async function sendTokensTriangular(
   }
 }
 
+async function sendTokensHubAndSpoke(
+  a: Network,
+  b: Network,
+  c: Network,
+  aActor: Key,
+  bActor: Key,
+  cActor: Key,
+  n: Nomad
+) {
+  const tokenFactory = getCustomToken();
+  const tokenOnA = await a.deployToken(
+    tokenFactory,
+    aActor.toAddress(),
+    "MyToken",
+    "MTK"
+  );
+
+  const token = {
+    domain: a.domain,
+    id: tokenOnA.address,
+  };
+
+  const ctx = n.getMultiprovider();
+
+  ctx.registerWalletSigner(a.name, aActor.toString());
+  ctx.registerWalletSigner(b.name, bActor.toString());
+
+  ctx.registerWalletSigner(c.name, cActor.toString());
+
+  // get 3 random amounts which will be bridged
+  const amount1 = randomTokens();
+  const amount2 = randomTokens();
+  const amount3 = randomTokens();
+
+
+  const [successA2C] = await sendTokensAndConfirm(
+    n,
+    a,
+    c,
+    token,
+    cActor.toAddress(),
+    [amount2, amount3, amount1]
+  );
+
+  if (!successA2C)
+    throw new Error(`Tokens transfer from ${a.name} to ${b.name} failed`);
+  console.log(`Tokens arrived at ${b.name}`);
+
+  const [successA2B] = await sendTokensAndConfirm(
+    n,
+    a,
+    b,
+    token,
+    bActor.toAddress(),
+    [amount1, amount2]
+  );
+
+  if (!successA2B)
+    throw new Error(`Tokens transfer from ${a.name} to ${b.name} failed`);
+  console.log(`Tokens arrived at ${b.name}`);
+
+  const [successB2A, tokenContract1] = await sendTokensAndConfirm(
+    n,
+    b,
+    a,
+    token,
+    new Key().toAddress(),
+    [amount1, amount2]
+  );
+
+  if (!successB2A)
+    throw new Error(`Tokens transfer from ${a.name} to ${b.name} failed`);
+  console.log(`Tokens arrived at ${b.name}`);
+
+  const [successC2A, tokenContract2] = await sendTokensAndConfirm(
+    n,
+    c,
+    a,
+    token,
+    new Key().toAddress(),
+    [amount2, amount3, amount1]
+  );
+
```

Comment:
thought: should we try B2C or C2B and require that it fails?

Reply:
added and it works